src/Groups/homomorphisms.jl

```diff
@@ -517,6 +524,29 @@ function isomorphism(::Type{T}, G::GAPGroup) where T <: Union{FPGroup, PcGroup,
    end::GAPGroupHomomorphism{typeof(G), T}
 end
 
+# If `G` is not a full pc group then switch to a full pc group in Oscar.
+function isomorphism(T::Type{PcGroup}, G::GAPGroup)
+   # Known isomorphisms are cached in the attribute `:isomorphisms`.
+   isos = get_attribute!(Dict{Type, Any}, G, :isomorphisms)::Dict{Type, Any}
+   return get!(isos, T) do
+     fun = _get_iso_function(T)
+     f = fun(G.X)::GapObj
+     @req f !== GAP.Globals.fail "Could not convert group into a group of type $T"
+     # The codomain of `f` can be a *subgroup* of a full pc group.
+     # In this situation, we have to switch to a full pc group.
+     C = GAP.Globals.Range(f)::GapObj
+     if GAPWrap.GeneratorsOfGroup(C) != GAP.Globals.FamilyPcgs(C)
```

Comment:
What about `pcp` groups, they don't use `FamilyPcgs` and probably needs separate code?